purchase_sale_inter_company/models/purchase_order_line.py

```diff
@@ -31,10 +31,13 @@ def create(self, vals_list):
         """Sync lines between an confirmed unlocked purchase and a confirmed unlocked
         sale order"""
         lines = super().create(vals_list)
+        not_allowed_states = ["cancel", "done"]
+        if self.env.context.get("allow_update_locked_sales", False):
+            not_allowed_states.remove("done")
         for order in lines.order_id.filtered(
```

Comment:
I would rather have this in a method that would allow overriding / extending the logic

Reply:
Thank @gurneyalex , I updated